ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -76,92 +72,75 @@ Web based editors like Word Online, Excel Online, Google Sheets, etc. may suppor
 
-```html
-<iframe
-    src="https://other-domain.com/will-monitor-clipboard.html"
-    allow="clipboard-read;"
->
-</iframe>
+```typescript
+interface ClipboardChangeEvent{
+  types: Array<string>; // MIME types available in the clipboard when the event was fired
+}
 ```
+The types member can be used to detect available data types present on the clipboard and then reflect the same on the UI as per [this scenario](#21-scenario-show-available-paste-formats-in-web-based-editors).
 
-#### 5.1.2 Approach 2 - No permission required
-Since no data is being sent as part of the clipboardchange event, it can be argued that we don't need any permission to simply know when clipboard contents change. This will simplify the user flow as they don't need to explicitly ask for permissions before listening to the event. It can also be noted that there is no identified incremental security risk in providing the "clipboardchange" event to web authors since this event doesn't allow the site to modify the system in any way - it simply tells when the clipboard contents have changed.
-
-##### Pros
-1. Simpler implementation and user experience
+### 4.3 Clipboard contents - Not available in event payload
 
-##### Cons
-1. This can cause user privacy violation - a site will be able to monitor the clipboard - a privileged OS component - without the user explicitly allowing it. (Though the author would only know that the clipboard changed, but wouldn't have any details about the payload)
+This API doesn't intend to provide any user clipboard contents as part of the event payload.
 
-#### 5.1.3 Conclusion
-We favour Approach 1 i.e. having clipboard-read permission required to listen to clipboardchange event, because it has more provisions which safeguards user privacy.
+### 4.4 Permissions and Interop - No user permission required 
 
-### 5.2 Page focus requirement
-As per the [current spec](https://www.w3.org/TR/clipboard-apis/#clipboard-event-clipboardchange), we should not fire "clipboardchange" event when a page is not is focus. This is in-line with the current behavior where async clipboard API is not accessible unless the given page is in focus. We do fire "clipboardchange" event when the page regains focus, incase the clipboard contents had changed when the page was out of focus. Note that even if the clipboard had changed multiple times while the page was out of focus, we will only fire a single "clipboardchange" event when the page regains focus. This is because the event is designed to indicate that the clipboard contents are different from what they were when the page lost focus, rather than tracking every individual change that occurred while the page was out of focus.
-
-#### 5.2.1 Approach 1 (Preferred) - Page required to be in focus to receive event
+When fired, this API intends to indicate that the clipboard has changed and provides the current MIME types present on the clipboard. Since no data exposed, there is no need for user permissions.
```

Comment:
```suggestion
When fired, this API indicates that the clipboard has changed and provides the current MIME types present on the clipboard. Since the actual contents of the clipboard are not exposed, there is no need for user permissions.
```